src/unix/darwin-stub.h

```diff
@@ -110,4 +111,9 @@ static const int kFSEventStreamEventFlagRootChanged = 32;
 static const int kFSEventStreamEventFlagUnmount = 128;
 static const int kFSEventStreamEventFlagUserDropped = 2;
 
+// Copied from https://opensource.apple.com/source/xnu/xnu-6153.101.6/libsyscall/wrappers/spawn/spawn_private.h.auto.html
+int     posix_spawnattr_set_uid_np(const posix_spawnattr_t * __restrict, uid_t) __API_AVAILABLE(macos(10.15), ios(13.0), tvos(13.0), watchos(6.0));
+int     posix_spawnattr_set_gid_np(const posix_spawnattr_t * __restrict, gid_t) __API_AVAILABLE(macos(10.15), ios(13.0), tvos(13.0), watchos(6.0));
+int     posix_spawnattr_set_groups_np(const posix_spawnattr_t * __restrict, int, gid_t * __restrict, uid_t) __API_AVAILABLE(macos(10.15), ios(13.0), tvos(13.0), watchos(6.0));
```

Comment:
```suggestion
int posix_spawnattr_set_uid_np(const posix_spawnattr_t*, uid_t);
int posix_spawnattr_set_gid_np(const posix_spawnattr_t*, gid_t);
int posix_spawnattr_set_groups_np(const posix_spawnattr_t*, int, gid_t*, uid_t);
```

Reply:
@bnoordhuis FYI after removing the `__API_AVAILABLE` macro, I get the following error when compiling this branch of libuv with Electron:

```
../../third_party/electron_node/deps/uv/src/unix/process.c:411:31: error: 'posix_spawn_file_actions_addchdir_np' is only available on macOS 10.15 or newer [-Werror,-Wunguarded-availability-new]
  if (options->cwd != NULL && posix_spawn_file_actions_addchdir_np(&actions, options->cwd))
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../../.electron_build_tools/third_party/Xcode/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/spawn.h:178:9: note: 'posix_spawn_file_actions_addchdir_np' has been marked as being introduced in macOS 10.15 here, but the deployment target is macOS 10.10.0
int     posix_spawn_file_actions_addchdir_np(posix_spawn_file_actions_t *,
        ^
../../third_party/electron_node/deps/uv/src/unix/process.c:411:31: note: enclose 'posix_spawn_file_actions_addchdir_np' in a __builtin_available check to silence this warning
  if (options->cwd != NULL && posix_spawn_file_actions_addchdir_np(&actions, options->cwd))
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Is there a preferred way in libuv of specifying the API availability? If not, no worries, it's easy enough to get Electron to compile w/ libuv by adding the `__API_AVAILABLE` macro back in as a patch in Electron.

Reply:
> Is there a preferred way in libuv of specifying the API availability?

In fact, yes :-) - `dlsym(RTLD_DEFAULT, "posix_spawn_file_actions_addchdir_np")`

Libuv generally uses `dlsym()` for things that might or might not be available at runtime. We try to avoid compile-time feature checks because of binary compatibility - a binary compiled on a newer system should, if at all possible, Just Work on an older system.